File: ContractTests/Http/modelBound/FixtureRateValidator.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.
import { ConceptValidator, validator } from '@cratis/arc.core';
import { FixtureRate } from './FixtureRate.js';
@validator(FixtureRate)
export class FixtureRateValidator extends ConceptValidator<FixtureRate> {
    constructor() {
        super();
        this.ruleFor(rate => rate.value).greaterThan(0).withMessage('Rate must be positive');
    }
}

File: ContractTests/DotNET/FixtureRateValidator.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Cratis.Arc.Validation;
using FluentValidation;

namespace HttpFixture;

/// <summary>
/// Validates the rate wherever it appears in the model graph.
/// </summary>
public sealed class FixtureRateValidator : ConceptValidator<FixtureRate>
{
    /// <summary>
    /// Declares the positive-rate rule.
    /// </summary>
    public FixtureRateValidator() => RuleFor(rate => rate.Value).GreaterThan(0m).WithMessage("Rate must be positive");
}
